### PR TITLE
[CONTINT-3558] Add e2e tests for APM origin detection on k8s

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR adds e2e tests for origin detection on APM.

### Motivation
Increase test coverage.

### Additional Notes
N/A

### Possible Drawbacks / Trade-offs
N/A

### Describe how to test/QA your changes
Run tests using `inv new-e2e-tests.run --targets=./tests/containers --run TestKindSuite`
```
=== RUN   TestKindSuite/TestTraceTCP
=== RUN   TestKindSuite/TestTraceUDP
```

### CODE REVIEW

**Comment by AliDatadog** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=auto

**Comment by AliDatadog** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=auto

File: test/new-e2e/tests/containers/k8s_test.go [hunk 1]
```diff
@@ -1036,3 +1039,52 @@ func (suite *k8sSuite) podExec(namespace, pod, container string, cmd []string) (
...
+				regexp.MustCompile(`^pod_name:` + kubeDeployment + `-[[:alnum:]]+-[[:alnum:]]+$`),
+				regexp.MustCompile(`^pod_phase:running$`),
+				regexp.MustCompile(`^short_image:apps-tracegen$`),
+			})
+			if err == nil {
+				break
+			}
+		}
+		require.NoErrorf(c, err, "Failed finding trace with proper tags and message")
+	}, 2*time.Minute, 10*time.Second, "Failed finding trace with proper tags and message")
```
[the remark this answers is not in the document]
Comment (AliDatadog) [reply test/new-e2e/tests/containers/k8s_test.go]: Good catch. I applied the suggestion. I suppose it is enough to only look at tags for now. Perhaps we can get some insights from #APM regarding what else we could check

**Review by L3n41c — APPROVED** [review]

File: test/new-e2e/tests/containers/k8s_test.go [hunk 2]
```diff
@@ -1036,3 +1039,55 @@ func (suite *k8sSuite) podExec(namespace, pod, container string, cmd []string) (
 
 	return stdoutSb.String(), stderrSb.String(), nil
 }
+
+func (suite *k8sSuite) TestTraceUDP() {
```
Comment (AliDatadog) [inline test/new-e2e/tests/containers/k8s_test.go:1043]:
```suggestion
func (suite *k8sSuite) TestTraceUDS() {
```

**Pull request merged**